src/client/java/teammates/client/scripts/DataMigrationForRubricQuestionsWithWeightsAttached.java

```diff
@@ -0,0 +1,112 @@
+package teammates.client.scripts;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.List;
+
+import com.googlecode.objectify.Key;
+
+import teammates.common.datatransfer.attributes.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.questions.FeedbackQuestionType;
+import teammates.common.datatransfer.questions.FeedbackRubricQuestionDetails;
+import teammates.storage.entity.FeedbackQuestion;
+
+/**
+ * Script to load Rubric questions, and if the question has weights attached, converts the one-dimensional
+ * rubric weights list to a two-dimensional weight list.
+ */
+public class DataMigrationForRubricQuestionsWithWeightsAttached extends DataMigrationBaseScript<Key<FeedbackQuestion>> {
+
+    public static void main(String[] args) throws IOException {
+        new DataMigrationForRubricQuestionsWithWeightsAttached().doOperationRemotely();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected boolean isPreview() {
+        return true;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected List<Key<FeedbackQuestion>> getEntities() {
+        return ofy().load().type(FeedbackQuestion.class).filter("questionType", FeedbackQuestionType.RUBRIC).keys().list();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected boolean isMigrationNeeded(Key<FeedbackQuestion> questionKey) {
+        FeedbackQuestion question = ofy().load().key(questionKey).now();
+        FeedbackQuestionAttributes attr = FeedbackQuestionAttributes.valueOf(question);
+        FeedbackRubricQuestionDetails fqd = (FeedbackRubricQuestionDetails) attr.getQuestionDetails();
+
+        try {
+            Field hasAssignedWeights = fqd.getClass().getDeclaredField("hasAssignedWeights");
+            hasAssignedWeights.setAccessible(true);
+            if (!hasAssignedWeights.getBoolean(fqd)) {
+                return false;
+            }
+
+            Field rubricWeightsForEachCell = fqd.getClass().getDeclaredField("rubricWeightsForEachCell");
+            rubricWeightsForEachCell.setAccessible(true);
+            List<List<Double>> newWeights = (List<List<Double>>) rubricWeightsForEachCell.get(fqd);
+
+            // Question contains legacy data if rubricWeightsForEachCell list is empty.
+            return newWeights.isEmpty();
+        } catch (ReflectiveOperationException e) {
+            System.out.println(e.getMessage());
+            return true;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void printPreviewInformation(Key<FeedbackQuestion> questionKey) {
+        // nothing to do.
```

Comment:
At least print something like:
`# of weighted rubric quesiton that need data migration:`

Reply:
I did not print this as I am only setting `isMigrationNeeded` to `true` for the rubric questions that have weights assigned, and the `rubricWeightsForEachCell` list is empty (which means there is legacy data). SO the `numberOfAffectedEntities` is the number of rubric questions that need data migration. 

Also `printPreviewInformation` is executed inside a loop, so wouldn't printing that be redundant?

https://github.com/TEAMMATES/teammates/blob/0c0419b5c092bfbf3c5e4a8a1f29535d594e7d3e/src/client/java/teammates/client/scripts/DataMigrationBaseScript.java#L77-L87

Reply:
Oh. Initially I was thinking at least we should print something to prove that the script is running :D

Reply:
I saved an output when I was running the script on my dev server. The output looks like this:

```
Going to connect to:localhost:8080
Running teammates.client.scripts.DataMigrationForRubricQuestionsWithWeightsAttached...
Preview: true
Total number of keys: 26
Number of affected keys: 8
Number of updated keys: 0
--- Remote operation completed ---
```

I haven't run the main migration script though.